softicar/platform
# PLAT-849 Fixed Hot-Deployment for AjaxFramework

- removed useless IAjaxFramework interface
- AjaxFramework.initialize no longer requires a full ServletContext
- extracted code to new project `platform-common-web`
- added new IWebServiceFactory interface for source code entry points
  - the service classes themselves are no source code entry points anymore
  - this simplifies the life-cycle of services, they are no longer singletons
- greatly simplified a lot of code

Test Plan:
- AjaxDomNodeServer
  - start AjaxDomNodeServer
  - open page, click and observe code works
  - change label of button in Eclipse and save
  - press F5
  - observe label of button is changed

The new Project:

![image](https://user-images.githubusercontent.com/83839745/168249540-b5c9cd13-21ba-4017-85b2-b006d75d3b8d.png)
